save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/ProjectView.kt

```diff
@@ -5,137 +5,407 @@
+import org.cqfn.save.frontend.utils.get
 import org.cqfn.save.frontend.utils.post
-import org.cqfn.save.repository.GitRepository
 
 import org.w3c.dom.HTMLInputElement
 import org.w3c.fetch.Headers
 import org.w3c.fetch.Response
+import org.w3c.files.Blob
+import org.w3c.files.BlobPropertyBag
+import org.w3c.files.File
+import org.w3c.files.get
+import org.w3c.xhr.FormData
 import react.RBuilder
 import react.RComponent
 import react.RProps
 import react.RState
 import react.child
 import react.dom.a
 import react.dom.button
+import react.dom.defaultValue
 import react.dom.div
 import react.dom.h1
 import react.dom.h2
 import react.dom.h6
 import react.dom.input
+import react.dom.label
 import react.dom.p
+import react.dom.span
+import react.dom.strong
 import react.setState
 
 import kotlinx.browser.window
 import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.await
 import kotlinx.coroutines.launch
 import kotlinx.html.ButtonType
 import kotlinx.html.InputType
+import kotlinx.html.hidden
+import kotlinx.html.id
 import kotlinx.html.js.onChangeFunction
 import kotlinx.html.js.onClickFunction
-
-/**
- * [RProps] for project view
- */
-@Suppress("MISSING_KDOC_CLASS_ELEMENTS")
-external interface ProjectProps : RProps {
-    var executionRequest: ExecutionRequest
-    var url: String?
-}
+import kotlinx.html.role
 
 /**
  * [RProps] retrieved from router
  */
 @Suppress("MISSING_KDOC_CLASS_ELEMENTS")
 external interface ProjectExecutionRouteProps : RProps {
-    var name: String
     var owner: String
-    var url: String?
+    var name: String
 }
 
 /**
  * [RState] of project view component
  */
 external interface ProjectViewState : RState {
     /**
-     * Whether error modal from backend is visible
+     * Binary file of project
+     */
+    var binaryFile: File?
+
+    /**
+     * Property file for project
+     */
+    var propertyFile: File?
+
+    /**
+     * Message of error
+     */
+    var errorMessage: String
+
+    /**
+     * Flag to handle error
      */
     var isErrorOpen: Boolean
 
     /**
-     * Whether error text from backend is visible
+     * Error label
      */
-    var errorText: String
+    var errorLabel: String
+
+    /**
+     * Flag to handle loading
+     */
+    var isLoading: Boolean
 }
 
 /**
  * A functional RComponent for project view
+ * Each modal opening call re render full page, that why we need to use state for all fields
  *
  * @return a functional component
  */
 @OptIn(ExperimentalJsExport::class)
 @JsExport
-class ProjectView : RComponent<ProjectProps, ProjectViewState>() {
-    private var pathToPropertyFile: String? = null
+@Suppress("CUSTOM_GETTERS_SETTERS")
+class ProjectView : RComponent<ProjectExecutionRouteProps, ProjectViewState>() {
+    private val testTypesList = listOf("Class", "Comment", "Function", "Variable", "Enum", "Space", "Style", "Another")
+    private var pathToProperty: String? = null
+    private var gitUrlProject: String? = null
```

Comment:
Please rename this variable to be more self-descriptive, like `gitUrlFromInputField` or something